.github/workflows/luarocks.yml

```
@@ -0,0 +1,24 @@
+name: Push to Luarocks
+
+on:
+  schedule:
+    - cron: "0 0 * * *"
+  workflow_dispatch:
+
+jobs:
+  luarocks-upload:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Get Version
+        run: |
+          if ! git describe --exact-match --tags; then
+            echo "LUAROCKS_VERSION=$(date +v%Y.%m.%d)" >> "${GITHUB_ENV}"
```

:thinking: I'm not sure if this is a good idea for luarocks, which expects SemVer.
If you release version `1.0.0` one day, then `luarocks install none-ls.nvim` will see `2024.4.2` as a greater version than the actual tagged release.

I would suggest to just publish an `scm` rockspec if you don't want to commit to SemVer tags yet.

```suggestion
            echo "LUAROCKS_VERSION='scm'" >> "${GITHUB_ENV}"
```

You could also simplify this by setting

```yaml
with:
    version: "scm"
```

and omitting the `LUAROCKS_VERSION` for now.

---

done in 1230cf2b46a7a8fada0d129433f7b456368a7d3a